thinkhazard/views/report.py

```diff
@@ -110,46 +142,36 @@ def report_json(request):
         raise HTTPBadRequest(detail='incorrect value for parameter '
                                     '"divisioncode"')
 
+    try:
+        resolution = float(request.params.get('resolution'))
+    except:
+        raise HTTPBadRequest(detail='invalid value for parameter "resolution"')
+
     hazard_type = request.matchdict.get('hazardtype', None)
 
-    division = DBSession.query(AdministrativeDivision) \
-        .join(AdminLevelType) \
-        .filter(AdministrativeDivision.code == division_code).one()
+    _filter = or_(AdministrativeDivision.code == division_code,
+                  AdministrativeDivision.parent_code == division_code)
 
-    _filter = None
-    if division.leveltype.mnemonic == u'REG':
-        _filter = AdministrativeDivision.code == division_code
-    else:
-        _filter = AdministrativeDivision.parent_code == division_code
+    simplify = StSimplify(AdministrativeDivision.geom, resolution / 2)
```

Comment:
The standard way to use a GeoAlchemy function is to use `func.ST_Simplify`. See https://github.com/geoalchemy/geoalchemy2/blob/9c4daa4432e74994c4ab7d7237cc9e9c9d2dfb9f/tests/test_functional.py#L260 for example.